### PR TITLE
feat(iroh-net): Log the crate version number

### PR DESCRIPTION
**Pull request by flub**
## Description

This adds a DEBUG-level log logging the crate version number.

## Breaking Changes

None

## Notes & open questions

This is not the first log message appearing, the MagicSock and a few
other pieces already have been logging by this time.  Also I'm not
particularly bound to this wording.  Maybe there are other things that
should be added to this line?

## Change checklist

- [x] Self-review.
- ~~[ ] Documentation updates following the [style guide](https://rust-lang.github.io/rfcs/1574-more-api-documentation-conventions.html#appendix-a-full-conventions-text), if relevant.~~
- ~~[ ] Tests if relevant.~~
- ~~[ ] All breaking changes documented.~~

### CODE REVIEW

File: iroh-net/src/endpoint.rs [hunk 1]
```diff
@@ -405,7 +405,7 @@ impl Endpoint {
             Arc::new(quinn::TokioRuntime),
         )?;
         trace!("created quinn endpoint");
-
+        debug!(version = env!("CARGO_PKG_VERSION"), "iroh Endpoint created");
```
Comment (matheus23) [inline iroh-net/src/endpoint.rs:408]: Open Q: Do we still need the trace log?

**Review by matheus23 — APPROVED** [review]

File: iroh-net/src/endpoint.rs (hunk 1 above)
Comment (dignifiedquire) [reply iroh-net/src/endpoint.rs:408]: what about commit version?

Comment (matheus23) [reply iroh-net/src/endpoint.rs:408]: Also I'm not sure if my linked answer would work with libraries. :thinking:

Comment (flub) [reply iroh-net/src/endpoint.rs:408]: > Open Q: Do we still need the trace log?

the trace logs distinguish magicsock creation from quinn endpoint creation.  i don't know right now if that's useful or not.  but it's trace so left it alone.

Comment (flub) [reply iroh-net/src/endpoint.rs:408]: git commit sounds useful, but I don't know how to do it - as shown by @matheus23 's responses.  happy to add it if there's a simple recipe though.

**Comment by flub** [comment]
> Did you test whether the environment variable reports the version of the crate, even if it's a deep transitive dependency? I.e. I'd like to avoid a case where it reports version "0.1.0", if you compile "my-crate v 0.1.0" which depends on "iroh-net v 0.26.0"

Pretty sure that's fine: https://doc.rust-lang.org/cargo/reference/environment-variables.html

**Pull request merged**